### PR TITLE
PD-2350 audit ts doc annotations

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Adds missing TSDoc annotations, updates type structures, and updates export style for better TSDoc generation

🎟 _Jira ticket:_ [PD-2350](https://jira.mongodb.org/browse/PD-2350)

## 🛠 Types of changes

- [x] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/button/src/Button.tsx [hunk 1]
```diff
@@ -14,71 +14,71 @@ import { ButtonContent } from './ButtonContent';
...
-      className,
-      as,
-      type,
-      ...rest
-    }: ButtonProps,
-    forwardRef,
-  ) {
-    const { usingKeyboard } = useUsingKeyboardContext();
-    const { darkMode } = useDarkMode(darkModeProp);
+export const Button: ExtendableBox<
```
Comment (TheSonOfThomp) [inline packages/button/src/Button.tsx:17]: Default export -> named export

**Comment by spark33** [comment]
There's a lot going on in this PR - this might be good to review as a team as a part of our TSDoc-updating grind session coming up.

File: packages/button/src/types.ts [hunk 2]
```diff
@@ -67,35 +67,47 @@ interface ButtonProps {
...
+
+  /**
+   * The content that will appear inside of the `<Button />` component.
+   */
   children?: React.ReactNode;
   /**
    * An icon glyph rendered before the button text.
-   *
-   * Default: `undefined`
+   * @type Leafygreen <Icon /> Element
```
Comment (TheSonOfThomp) [inline packages/button/src/types.ts:101]: *`component`

Comment (TheSonOfThomp) [reply packages/button/src/types.ts:101]: Mention using `createIconComponent` in docs

Comment (TheSonOfThomp) [reply packages/button/src/types.ts:101]: TODO: consider locking this down to be LG Icon in TS

File: scripts/parse-tsdocs.ts [hunk 3]
```diff
@@ -85,12 +98,12 @@ function parseDocs(componentName: string): void {
     const componentFileNames = parseFileNames(componentDir);
 
     const docs: Array<CustomComponentDoc> = uniqBy(
-      parse(componentFileNames, TSDocOptions)
+      Parser.parse(componentFileNames)
```
Comment (TheSonOfThomp) [inline scripts/parse-tsdocs.ts:101]: Fix the 5x loops

File: packages/checkbox/src/Check/index.tsx [hunk 4]
```diff
@@ -36,6 +36,10 @@ const checkIconColor: Record<Theme, Record<'default' | 'disabled', string>> = {
   },
 };
 
+/**
+ * @internal
+ * @returns
```
Comment (TheSonOfThomp) [inline packages/checkbox/src/Check/index.tsx:41]: remove

File: packages/combobox/src/ComboboxMenu/ComboboxMenu.tsx [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import React, { useContext, useMemo } from 'react';
+import React, { forwardRef, useContext, useMemo } from 'react';
```
Comment (TheSonOfThomp) [inline packages/combobox/src/ComboboxMenu/ComboboxMenu.tsx:1]: Not necessary anymore

File: packages/icon/src/Icon.tsx [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import { createIconComponent } from './createIconComponent';
+import { glyphs } from './glyphs';
+
+export const Icon = createIconComponent(glyphs);
```
Comment (TheSonOfThomp) [inline packages/icon/src/Icon.tsx:4]: consider using `.ts`

File: packages/inline-definition/src/InlineDefinition.tsx [hunk 7]
```diff
@@ -50,19 +50,20 @@ const triggerElementModeStyles: Record<Theme, string> = {
   `,
 };
 
-type InlineDefinitionProps = Partial<TooltipProps> & {
+interface InlineDefinitionProps extends TooltipProps {
```
Comment (TheSonOfThomp) [inline packages/inline-definition/src/InlineDefinition.tsx:53]: Return `Partial<>`

File: packages/menu/src/MenuItem.tsx [hunk 8]
```diff
@@ -29,7 +33,7 @@ import { Size } from './types';
 import MenuContext from './MenuContext';
 
 const menuItemContainer = createDataProp('menu-item-container');
-interface BaseMenuItemProps {
+interface MenuItemProps extends HTMLElementProps<'button'> {
```
Comment (TheSonOfThomp) [inline packages/menu/src/MenuItem.tsx:36]: Use `BoxProps`

Comment (TheSonOfThomp) [reply packages/menu/src/MenuItem.tsx:36]: Make sure `as` prop works

File: packages/menu/src/SubMenu.tsx [hunk 9]
```diff
@@ -274,7 +279,7 @@ const subItemThemeStyle: Record<Theme, string> = {
 
 const subMenuItemHeight = 32;
 
-interface SubMenuProps {
+interface SubMenuProps extends HTMLElementProps<'button'> {
```
Comment (TheSonOfThomp) [inline packages/menu/src/SubMenu.tsx:282]: Same here: make sure `as` prop works

File: packages/banner/src/types.ts [hunk 10]
```diff
@@ -16,7 +16,8 @@ export interface BannerProps extends HTMLElementProps<'div', never> {
   /**
    * Sets the variant for the Banner
    *
-   * Default: `'info'`
+   * @required
```
Comment (TheSonOfThomp) [inline packages/banner/src/types.ts:19]: Leave out `@required` annotations. Just ensure defaults are appropriately set

File: packages/badge/src/Badge.tsx [hunk 11]
```diff
@@ -20,7 +20,7 @@ interface BadgeProps {
   /**
    * The Badge's style variant
    *
-   * Default: `'lightgray'`
+   * @default: 'lightgray'
```
Comment (TheSonOfThomp) [inline packages/badge/src/Badge.tsx:23]: Remove `@default:` with colons

**Pull request merged**